data-pipeline/src/trace_exporter.rs

```diff
@@ -214,32 +253,173 @@ impl TraceExporter {
+                    bucket_size,
+                }));
+        }
+    }
+
+    /// Check for a new state of agent_info and update the trace exporter if needed
+    fn check_agent_info(&self) {
+        dbg!(self.agent_info.load().clone());
+        if let Some(agent_info) = self.agent_info.load().as_deref() {
+            if Some(agent_info.state_hash.as_str())
+                != self
+                    .previous_info_state
+                    .load()
+                    .as_deref()
+                    .map(|s| s.as_str())
+            {
+                match &**self.client_side_stats.load() {
+                    StatsComputationStatus::Disabled => {}
+                    StatsComputationStatus::DisabledByAgent { .. } => {
+                        if agent_info.info.client_drop_p0s.is_some_and(|v| v) {
+                            // Client-side stats is supported by the agent
+                            let status = self.start_stats_computation(
+                                agent_info
+                                    .info
+                                    .span_kinds_stats_computed
+                                    .clone()
+                                    .unwrap_or_else(|| {
+                                        DEFAULT_STATS_ELIGIBLE_SPAN_KINDS.map(String::from).to_vec()
+                                    }),
+                                agent_info.info.peer_tags.clone().unwrap_or_default(),
+                            );
+                            match status {
+                                Ok(()) => info!("Client-side stats enabled"),
+                                Err(_) => error!("Failed to start stats computation"),
+                            }
+                        } else {
+                            info!("Client-side stats computation has been disabled by the agent")
+                        }
+                    }
+                    StatsComputationStatus::Enabled {
+                        stats_concentrator,
+                        cancellation_token: _,
+                        exporter_handle: _,
+                    } => {
+                        if agent_info.info.client_drop_p0s.is_some_and(|v| v) {
+                            let mut concentrator = stats_concentrator.lock().unwrap();
+                            concentrator.set_span_kinds(
+                                agent_info
+                                    .info
+                                    .span_kinds_stats_computed
+                                    .clone()
+                                    .unwrap_or_default(),
+                            );
+                            concentrator.set_peer_tags(
+                                agent_info.info.peer_tags.clone().unwrap_or_else(|| {
+                                    DEFAULT_STATS_ELIGIBLE_SPAN_KINDS.map(String::from).to_vec()
+                                }),
```

Comment:
So why does this code differ from the code above, where the span kinds use this default, but here it's added to the peer tags 🤷🏼‍♂️

Reply:
That's just me going crazy, I fixed it.